Automattic/jetpack
# Add endpoints to retrieve and delete a site's purchase token

This PR introduces two new endpoints under the `jetpack/v4` namespace which are meant to be used as part of the new logged-out visitor checkout flow. The purpose of these endpoints is to allow clients to retrieve and delete a site's purchase token. 

These changes come from https://github.com/Automattic/jetpack/pull/19633 so if you want to know more about them, I encourage you to read that PR's description. 

#### Changes proposed in this Pull Request:
* Add two `jetpack/v4` endpoints to retrieve and delete a site's purchase token.
* Add a Jetpack Option called 'purchase_token' that stores a token for logged-out user purchases.
* Enable site-level authentication for REST API endpoints.

#### Jetpack product discussion
* Project Thread: Logged Out Visitor Checkout.
* Asana Task: 1200152453830945-as-1200222681647748

#### Does this pull request change what data or activity we track or use?
* No, it doesn't.

#### Testing instructions:
* Run tests with `yarn docker:phpunit --filter=WP_Test_Jetpack_REST_API_endpoints` and make sure all of them pass.
* Use 2920a-pb to test these endpoints with both a blog and user token. You can add a new method to that class like this:
```
	public function get_purchase_token() {
		return $this->call_core_api( 'GET', '/jetpack/v4/purchase-token/' );
	}
```
* The method then can be called with `$t->get_purchase_token();`.
